### PR TITLE
Fixed and added javadoc comments for existing classes

### PR DESCRIPTION
**Pull request by legacycode**
# Description

I fixed some javadoc tags and added documentation to existing classes.

## Type of change

- Documentation Fix

# How Has This Been Tested?

- mvn clean install

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/com/iota/iri/service/dto/GetNeighborsResponse.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
      *  numberOfAllTransactions, numberOfRandomTransactionRequests, 
      *  numberOfNewTransactions, numberOfInvalidTransactions, numberOfSentTransactions
      * 
-     * @see {@link com.iota.iri.service.dto.GetNeighborsResponse.Neighbor}
+     * @see com.iota.iri.service.dto.GetNeighborsResponse.Neighbor
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/service/dto/GetNeighborsResponse.java]: @kwek20 
Make sure that this won't mess up your mdx generation. Basically the fix is correct. Thanks

**Comment by kwek20** [comment]
@GalRogozinski I Ran MDX, Runs fine! Docs are normal.
`@see` of a response is not used in the MDX generation.

**Comment by kwek20** [comment]
@GalRogozinski 

`@see` will be used in the v2 of the Doclet to pickup fields of referenced class, so the list is no longer required when we run it. Since we're not ready with that yet, Ill remove the list in the PR, this is okay for now.

**Comment by legacycode** [comment]
@GalRogozinski could you please merge this PR or are there any task for me?

**Comment by GalRogozinski** [comment]
Hey @legacycode 

Sorry for not responding, we were at a race to release Local Snapshots before Christmas (as you have noticed did not happen).

After Local Snapshots will be released we will start to sweep over the PRs and merge them.
I would have merged this now if it wasn't for the conflict to be honest.

If you want some small tasks you can do #1231, #1232,  #930, or #1221.
If you want a more challenging task do #1090 (best as an IXI).

I can think of other tasks but they need to be properly groomed before.
We will talk after Christmas :-)

**Comment by legacycode** [comment]
@GalRogozinski will this be merged? its from September and approved.

**Comment by jakubcech** [comment]
@kwek20 is this good to go from your PoV (yes I see that you approved above 😺)?  And any dependencies with regards to your doclet work, like #1155?

**Comment by kwek20** [comment]
@jakubcech This is still valid, as were still waiting for the next release of maven-javadoc-plugin before releasing the new doclet. Can be merged!

**Pull request merged**